test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java

```diff
@@ -1721,4 +1726,44 @@ public void testRegisterHandlerTwice() {
         serviceA.registerRequestHandler("action1", TestRequest::new, randomFrom(ThreadPool.Names.SAME, ThreadPool.Names.GENERIC),
             (request, message) -> {throw new AssertionError("boom");});
     }
+
+    public void testTimeoutPerConnection() throws IOException {
```

Comment:
I like this test, maybe just add an `assumeTrue(Constants.LINUX)` (even then I think we want >= 2.2 kernels for the modern backlog behavior, but that's mostly a given these days anyway so I'm not going to lose sleep over it)?